build.assets/tooling/cmd/build-apt-repos/main.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"container/list"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/seqsense/s3sync"
+)
+
+func main() {
+	// artifactPath := "/artifacts"
+	// bucketName := "aptly-test20220223185137234100000001"
+	// localBucketPath := "/bucket"
+	// majorVersion := "vTODO"
+	// releaseChannel := "stable"
+	supportedOSs := map[string][]string{
+		"debian": {
+			"stretch",
+			"buster",
+			"bullseye",
+			"bookwork",
+			"trixie",
+		},
+		"ubuntu": {
+			"xenial",
+			"yakkety",
+			"zesty",
+			"artful",
+			"bionic",
+			"cosmic",
+			"disco",
+			"eoan",
+			"focal",
+			"groovy",
+		},
+	}
+
+	args := parseFlags()
+
+	awsSession := session.Must(session.NewSession())
+	aptly := SetupAptly()
+
+	downloadExistingRepo(awsSession, *args.bucketName, *args.localBucketPath)
+
+	repos := list.New()
```

Comment:
Thanks for pointing this out. I've switched from lists to slices everywhere. Same thing with channels.

I had originally used linked lists everywhere because they are more performance in this use case, but it made the code a bit messier. Same thing with channels - by returning items as they're computed, they don't need to be stored while the rest are computed.